core/types/auxpow_coinbase_utils.go

```diff
@@ -0,0 +1,687 @@
+			currentHash = btcblockchain.HashMerkleBranches(&currentHash, &sibling)
+		}
+		return currentHash
+	case Scrypt:
+		// Start with the transaction hash
+		currentHash := ltcchainhash.Hash(AuxPowTxHash(powId, coinbaseTx))
+
+		// For coinbase (index 0), we always take the right branch
+		// and our hash goes on the left
+		for _, siblingBytes := range merkleBranch {
+			var sibling ltcchainhash.Hash
+			copy(sibling[:], siblingBytes)
+
+			// Since we're at index 0 (coinbase), we're always the left child
+			currentHash = ltcblockchain.HashMerkleBranches(&currentHash, &sibling)
+		}
+		return currentHash
+	case SHA_BCH:
+		// Start with the transaction hash
+		currentHash := bchchainhash.Hash(AuxPowTxHash(powId, coinbaseTx))
+
+		// For coinbase (index 0), we always take the right branch
+		// and our hash goes on the left
+		for _, siblingBytes := range merkleBranch {
+			var sibling bchchainhash.Hash
+			copy(sibling[:], siblingBytes)
+
+			// Since we're at index 0 (coinbase), we're always the left child
+			currentHash = *bchblockchain.HashMerkleBranches(&currentHash, &sibling)
+		}
+		return currentHash
+	default:
+		return [common.HashLength]byte{}
+	}
+}
+
+// ExtractMerkleBranch extracts the merkle branch for the coinbase (index 0)
+// from a complete merkle tree
+func ExtractMerkleBranch(merkleTree []*chainhash.Hash, txCount int) [][]byte {
+	if len(merkleTree) == 0 || txCount == 0 {
+		return nil
+	}
+
+	var branch [][]byte
+
+	// blockchain.BuildMerkleTreeStore lays out the tree level-by-level using a
+	// power-of-two leaf width. Determine that width so we can walk the levels.
+	width := 1
+	for width < txCount {
+		width <<= 1
+	}
+
+	index := 0
+	levelOffset := 0
+	levelWidth := width
+
+	for levelWidth > 1 {
+		siblingIndex := index ^ 1
+		siblingPos := levelOffset + siblingIndex
+		if siblingIndex >= levelWidth || siblingPos >= len(merkleTree) || merkleTree[siblingPos] == nil {
+			siblingPos = levelOffset + index
+		}
+		branch = append(branch, merkleTree[siblingPos][:])
+
+		levelOffset += levelWidth
+		index >>= 1
+		levelWidth >>= 1
+	}
+
+	return branch
+}
+
+// encodeHeightForBIP34 encodes block height in minimal little-endian format
+// This matches Bitcoin's CScriptNum serialization used for BIP34 compliance.
+// The height must be encoded with the minimum number of bytes required.
+func encodeHeightForBIP34(height uint32) []byte {
+	if height == 0 {
+		return []byte{}
+	}
+
+	// Convert to little-endian bytes, removing leading zeros
+	result := make([]byte, 0, 4)
+	for height > 0 {
+		result = append(result, byte(height&0xff))
+		height >>= 8
+	}
+
+	// If the most significant bit is set, add a zero byte to indicate positive number
+	// This is part of Bitcoin's CScriptNum format to distinguish positive/negative
+	if len(result) > 0 && (result[len(result)-1]&0x80) != 0 {
+		result = append(result, 0x00)
+	}
+
+	return result
+}
+
+// TODO: Replace with a method from btcd/ltcd/bchd if available
```

Comment:
Remove TODO unless you really plan on doing this

Reply:
will remove this, i dont think  i am going to do it